pkg/controllers/multiclusterservice/endpointslice_dispatch_controller.go

```diff
@@ -0,0 +1,279 @@
+
+	networkingv1alpha1 "github.com/karmada-io/karmada/pkg/apis/networking/v1alpha1"
+	workv1alpha1 "github.com/karmada-io/karmada/pkg/apis/work/v1alpha1"
+	"github.com/karmada-io/karmada/pkg/events"
+	"github.com/karmada-io/karmada/pkg/util"
+	"github.com/karmada-io/karmada/pkg/util/fedinformer/genericmanager"
+	"github.com/karmada-io/karmada/pkg/util/helper"
+	"github.com/karmada-io/karmada/pkg/util/names"
+)
+
+// EndpointsliceDispatchControllerName is the controller name that will be used when reporting events.
+const EndpointsliceDispatchControllerName = "endpointslice-dispatch-controller"
+
+var (
+	endpointSliceGVK = discoveryv1.SchemeGroupVersion.WithKind("EndpointSlice")
+)
+
+type EndpointsliceDispatchController struct {
+	client.Client
+	EventRecorder   record.EventRecorder
+	RESTMapper      meta.RESTMapper
+	InformerManager genericmanager.MultiClusterInformerManager
+}
+
+// Reconcile performs a full reconciliation for the object referred to by the Request.
+func (c *EndpointsliceDispatchController) Reconcile(ctx context.Context, req controllerruntime.Request) (controllerruntime.Result, error) {
+	klog.V(4).Infof("Reconciling Work %s", req.NamespacedName.String())
+
+	work := &workv1alpha1.Work{}
+	if err := c.Client.Get(ctx, req.NamespacedName, work); err != nil {
+		if apierrors.IsNotFound(err) {
+			return controllerruntime.Result{}, nil
+		}
+		return controllerruntime.Result{Requeue: true}, err
+	}
+
+	if !work.DeletionTimestamp.IsZero() {
+		if err := c.cleanupEndpointSliceFromConsumerClusters(ctx, work); err != nil {
+			klog.Errorf("Failed to cleanup EndpointSlice from consumer clusters for work %s/%s", work.Namespace, work.Name)
+			return controllerruntime.Result{Requeue: true}, err
+		}
+		return controllerruntime.Result{}, nil
+	}
+
+	if !helper.IsWorkContains(work.Spec.Workload.Manifests, endpointSliceGVK) {
+		return controllerruntime.Result{}, nil
+	}
+
+	mcsName := util.GetLabelValue(work.Labels, util.ServiceNameLabel)
+	mcsNS := util.GetLabelValue(work.Labels, util.ServiceNamespaceLabel)
+	mcs := &networkingv1alpha1.MultiClusterService{}
+	if err := c.Client.Get(ctx, types.NamespacedName{Namespace: mcsNS, Name: mcsName}, mcs); err != nil {
+		if apierrors.IsNotFound(err) {
+			klog.Warningf("MultiClusterService %s/%s is not found", mcsNS, mcsName)
+			return controllerruntime.Result{}, nil
+		}
+		return controllerruntime.Result{Requeue: true}, err
+	}
+
+	var err error
+	defer func() {
+		if err != nil {
+			_ = c.updateEndpointSliceSynced(mcs, metav1.ConditionFalse, "EndpointSliceSyncFailed", err.Error())
+			c.EventRecorder.Eventf(mcs, corev1.EventTypeWarning, events.EventReasonDispatchEndpointSliceFailed, err.Error())
+			return
+		}
+		_ = c.updateEndpointSliceSynced(mcs, metav1.ConditionTrue, "EndpointSliceSyncSucceed", "EndpointSlice are synced successfully")
+		c.EventRecorder.Eventf(mcs, corev1.EventTypeWarning, events.EventReasonDispatchEndpointSliceFailed, "EndpointSlice are synced successfully")
```

Comment:
The event does not need to be reported, and the event reason 'EventReasonDispatchEndpointSliceFailed' is incorrect.

Reply:
Yes, the event type and reason is incorrect, but here needs a positive event `EventReasonDispatchEndpointSliceSucceed` to cancel the previous `DispatchEndpointSliceFailed`.